src/javascript/index.js

```diff
@@ -0,0 +1,12 @@
+import {registry} from '@jahia/ui-extender';
+import registrations from './serverSettings';
+import i18next from 'i18next';
+
+registry.add('callback', 'serverSettings', {
+    targets: ['jahiaApp-init:50'],
+    callback: async () => {
+        await i18next.loadNamespaces('serverSettings');
```

Comment:
Do we need to await for loadNamespaces() before running registrations ? It could actually be an issue, as the callbacks are not treated as promises by jahia-ui-root - it just calls "callback" and do not expect the result to be a promise as it is here - and so the app could be loaded before the registration (here, after an await) happen.

Reply:
Yes, we do need to wait for them to load, that is if they take a while to load (I simulated with a timeout) and I run my registrations beforehand there is no translation until the next rerender. Technically even that "useEffect" trick that we do elsewhere won't work because useTranslation hooks are not notified when namespace is loaded. I looked at the code in jahia-ui-root and you're right it doesn't handle promises. What do you think about adding `Promise.all()` to make sure that callbacks have done what they need to do? I also just realized that what we do is not technically a callback as it runs before we render the app.

Reply:
I don't see any issue with running registration while you don't have any namespaces loaded, since you don't do any call anymore to i18next during registration. Translation are only called during render, so it's more at that time that we need the i18n to be available. I think content editor had the same issue, and they came out with another solution (which I don't completely understand and actually don't like neither) : https://github.com/Jahia/content-editor/blob/master/src/javascript/register.jsx - can you check with @FBerthelot ?

Reply:
I added Promise.all() after our call today, there is a PR here https://github.com/Jahia/jahia-ui-root/pull/42. I will merge this PR as it is blocking further work.